### PR TITLE
kotlin-multiplatform migration

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: gradle/publish.gradle [hunk 1]
```diff
@@ -1,8 +1,73 @@
+// Configures publishing of Maven artifacts to Bintray
+
+apply plugin: 'maven'
+apply plugin: 'maven-publish'
+apply plugin: 'com.jfrog.bintray'
+
+apply from: project.rootProject.file('gradle/pom.gradle')
+
+// ------------- tasks
+task stubSources(type: Jar) {
```
Comment (e5l) [inline gradle/publish.gradle]: delete

File: benchmarks/build.gradle [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
+    jmh rootProject
+    jmh rootProject.kotlin.targets.jvm.compilations.main.output.allOutputs
+    jmh rootProject.kotlin.targets.jvm.compilations.test.output.allOutputs
+    jmhCompile rootProject
+    jmhCompile rootProject.kotlin.targets.jvm.compilations.test.output.allOutputs
+
+    compile group: 'org.openjdk.jmh', name: 'jmh-core', version: '1.21'
 
+    jmhCompileClasspath rootProject
+    jmhCompileClasspath rootProject.kotlin.targets.jvm.compilations.main.output.allOutputs
```
Comment (cy6erGn0m) [inline benchmarks/build.gradle]: This is completely insane :scream:

File: build.gradle [hunk 3]
```diff
@@ -1,172 +1,91 @@
...
 
-                artifact sourceJar
+            kotlin.srcDir "$platform/$srcDir"
 
-                artifact emptyJar {
-                    classifier 'javadoc'
+            if (!skipPublish.contains(project.name)) {
+                languageSettings {
+                    progressiveMode = true
+                    experimentalAnnotations.each { useExperimentalAnnotation(it) }
```
Comment (cy6erGn0m) [inline build.gradle]: Wasn't `useExperimentalAnnotation` removed?

File: build.gradle [hunk 4]
```diff
@@ -1,172 +1,91 @@
...
 
-                configurePom(pom)
+        configure(targets) {
+            def targetName = it.name
+            compilations.all { compilation ->
+                def compileTask = tasks.getByName(compilation.compileKotlinTaskName)
+                compileTask.kotlinOptions.freeCompilerArgs += [
+                    "-progressive",
+                    "-Xuse-experimental=" +
+                        "kotlin.Experimental," +
```
Comment (cy6erGn0m) [inline build.gradle]: Shouldn't we concatenate it from the list from `gradle/experimental.gradle` ?

File: gradle/ide.gradle [hunk 5]
```diff
@@ -0,0 +1,13 @@
+task prepareIdeInterop(type: Copy) {
+    if (project.ext.isLinuxHost) {
```
Comment (cy6erGn0m) [inline gradle/ide.gradle]: What about windows?

**Pull request merged**